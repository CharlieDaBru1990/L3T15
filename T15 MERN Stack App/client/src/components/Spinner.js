import React from "react";

// The code provided defines a functional component called Spinner.
// Functional components are a simpler way to write React components compared to class components.
// The Spinner component returns JSX code that represents the visual appearance of the spinner

function Spinner() {
  return (
    <div className="fixed inset-0 bg-black opacity-70 flex items-center justify-center z-[9999]">
      <div className="h-10 w-10 border-4 border-gray-200 border-solid border-t-transparent rounded-full animate-spin"></div>
    </div>
  );
}

export default Spinner;
