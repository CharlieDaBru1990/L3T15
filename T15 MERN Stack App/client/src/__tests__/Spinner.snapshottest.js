// Spinner.test.js
import React from "react";
import { render } from "@testing-library/react";
import Spinner from "../components/Spinner";

test("Spinner component snapshot", () => {
  const { asFragment } = render(<Spinner />);
  expect(asFragment()).toMatchSnapshot();
});
