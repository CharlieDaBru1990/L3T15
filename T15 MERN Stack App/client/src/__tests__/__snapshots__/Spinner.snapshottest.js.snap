// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 bg-black opacity-70 flex items-center justify-center z-[9999]"
  >
    <div
      class="h-10 w-10 border-4 border-gray-200 border-solid border-t-transparent rounded-full animate-spin"
    />
  </div>
</DocumentFragment>
`;
